### PR TITLE
Add DropEventsResponseCode for eventshub

### PR DESCRIPTION
**Pull request by xinydev**
# Changes

- :gift: Add a new Option(DropEventsResponseCode) for eventshub to specify the HTTP response code
 when events should be dropped

/kind <kind>

Fixes https://github.com/knative-sandbox/reconciler-test/issues/211

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
Thank you!

**Comment by xinydev** [comment]
> Thank you!

Could you please add a approved label for this PR?

**Comment by pierDipi** [comment]
I'm not an approver in this repo.

/cc @slinkydeveloper

File: test/example/prober_feature.go [hunk 1]
```diff
@@ -84,8 +85,8 @@ func ProberFeatureWithDrop() *feature.Feature {
 			for _, event := range events {
 				switch event.Sent.SentId {
 				case "1", "2", "3", "4", "5":
-					if event.Response.StatusCode/100 != 4 {
-						t.Errorf("For %s, expected 4xx response, got %d", event.Sent.SentId, event.Response.StatusCode)
+					if event.Response.StatusCode != 429 {
```
Comment (n3wscott) [inline test/example/prober_feature.go:88]: this change should be reverted, we sent to more than just the eventshub receiver.

Comment (n3wscott) [reply test/example/prober_feature.go:88]: my mistake, this is the example test :D

**Review by n3wscott — APPROVED** [review]

**Pull request merged**